v2/errors.go

```diff
@@ -5,19 +5,181 @@
 
 package v2
```

Comment:
Can we move this out of V2 since it could/should be used in all areas not just the V2 API?

Reply:
where do you think it is appropriate?  how about the `errors.go` in `models` package?
should we create an `errors` package?

Reply:
There is already an `errors.go` in `models` and this is not really a `model`, so yea and `errors` package. Then we have what they call stuttering in the names. i.e. `errors.EdgexError` & `errors.CommonEdgeXError`. Naming guidelines say they should be something like `errors.Edgex` and `errors.CommonEdgex`.

Reply:
renamed the package, interface, struct, and functions.